### PR TITLE
Fixing CA1081

### PR DESCRIPTION
**Pull request by jamdavi**

### CODE REVIEW

File: build.ps1 [hunk 1]
```diff
@@ -120,7 +120,7 @@ Function BuildProject($path, $message)
         }
     }
 
-    & dotnet build $projectPath --verbosity $verbosity --configuration $configuration
+    & dotnet build $projectPath --verbosity $verbosity --configuration $configuration -logger:"C:\Users\jamdavi\source\repos\BasicFileLogger\BasicFileLogger\bin\Debug\BasicFileLogger.dll;out.log;CA1801"
```
Comment (drwill-ms) [inline build.ps1:123]: I don't think we want your path in this file

Comment (jamdavi) [reply build.ps1:123]: Doh

File: iothub/device/src/ModernDotNet/Common/IOThreadTimerSlim.cs [hunk 2]
```diff
@@ -17,6 +18,7 @@ private void CreateTimer()
             this.timer = new Timer((obj) => callback(obj), callbackState, TimeSpan.FromMilliseconds(-1), TimeSpan.FromMilliseconds(-1));
         }
 
+        [SuppressMessage("Usage", "CA1801:Review unused parameters", Justification = "Unknown if this is used by customers so we should leave as is to preserve the API.")]
```
Comment (drwill-ms) [inline iothub/device/src/ModernDotNet/Common/IOThreadTimerSlim.cs:21]: internal class => not used by customers

File: iothub/device/src/Transport/AmqpIoT/AmqpIoTMessageConverter.cs [hunk 3]
```diff
@@ -46,6 +47,7 @@ public static Message AmqpMessageToMessage(AmqpMessage amqpMessage)
             return message;
         }
 
+        [SuppressMessage("Usage", "CA1801:Review unused parameters", Justification = "Unknown if this is used by customers so we should leave as is.")]
```
Comment (drwill-ms) [inline iothub/device/src/Transport/AmqpIoT/AmqpIoTMessageConverter.cs:50]: same, internal class => not used by customers

File: iothub/device/src/Common/ExceptionTrace.cs [hunk 4]
```diff
@@ -208,16 +205,6 @@ internal void BreakOnException(Exception exception)
 #endif
         }
 
-        [MethodImpl(MethodImplOptions.NoInlining)]
```
Comment (jamdavi) [inline iothub/device/src/Common/ExceptionTrace.cs:211]: Does nothing, not used anywhere...

File: iothub/device/src/Common/Fx.cs [hunk 5]
```diff
@@ -258,27 +259,6 @@ private static bool TryGetDebugSwitch(string name, out object value)
 
 #endif // DEBUG
 
-        [SuppressMessage(FxCop.Category.Design, FxCop.Rule.DoNotCatchGeneralExceptionTypes,
```
Comment (jamdavi) [inline iothub/device/src/Common/Fx.cs:261]: After removing TraceUnhandled (which did nothing). This method did nothing...

File: iothub/device/src/Common/ExceptionTrace.cs [hunk 6]
```diff
@@ -90,18 +91,13 @@ public void TraceHandled(Exception exception, string catchLocation, EventTraceAc
             this.BreakOnException(exception);
         }
 
-        public void TraceUnhandled(Exception exception)
```
Comment (jamdavi) [inline iothub/device/src/Common/ExceptionTrace.cs:93]: Does nothing

File: iothub/device/src/Common/Singleton.cs [hunk 7]
```diff
@@ -39,7 +40,7 @@ public Task OpenAsync(TimeSpan timeout, CancellationToken cancellationToken)
             return GetOrCreateAsync(timeout, cancellationToken);
         }
 
-        public Task CloseAsync(CancellationToken cancellationToken)
+        public Task CloseAsync()
```
Comment (drwill-ms) [inline iothub/device/src/Common/Singleton.cs:43]: Why not remove this impl?

**Review by drwill-ms — APPROVED** [review]

File: iothub/device/src/Common/Singleton.cs (hunk 7 above)
Comment (jamdavi) [reply iothub/device/src/Common/Singleton.cs:43]: Even though this really isn't async all the way down it seems better to be consistent we have OpenAsync, GetOrCreateAsync, etc..

File: iothub/device/src/Common/ExceptionTrace.cs [hunk 8]
```diff
@@ -73,6 +73,7 @@ public ObjectDisposedException ObjectDisposed(string message)
             return TraceException<ObjectDisposedException>(new ObjectDisposedException(null, message), TraceEventType.Error);
         }
 
+        [SuppressMessage("Usage", "CA1801:Review unused parameters", Justification = "Unused parameters are inside of the DEBUG compilation flag.")]
```
Comment (abhipsaMisra) [inline iothub/device/src/Common/ExceptionTrace.cs:76]: nit: worth mentioning here what the unused parameter is.

File: iothub/device/src/Common/Singleton.cs (hunk 7 above)
Comment (abhipsaMisra) [inline iothub/device/src/Common/Singleton.cs:43]: Q - why not throw if cancellation was requested? Does the upstream call not have a cancellation token available?

Comment (jamdavi) [reply iothub/device/src/Common/Singleton.cs:43]: One thing I thought about here was we only Dispose the object and return a task complete. If we cancel this should it be after the dispose? If so do we really want to throw an exception if we're disposing the object?

Comment (abhipsaMisra) [reply iothub/device/src/Common/Singleton.cs:43]: You are correct, throwing an exception after disposing makes no sense.
My idea is to to check for cancellation before performing any processing operation; that way, we do not dispose if a cancellation has been requested.

Comment (jamdavi) [reply iothub/device/src/Common/Singleton.cs]: Upon further review this is not used by anyone and it's not public so it's gone.

**Review by drwill-ms — APPROVED** [review]

**Pull request merged**